charon/src/bin/charon-driver/translate/translate_crate_to_ullbc.rs

```diff
@@ -360,6 +361,29 @@ impl<'tcx, 'ctx> TranslateCtx<'tcx, 'ctx> {
     }
 }
 
+/// Read and return the content of a source file.
+fn read_file_content(filename: &FileName) -> Option<Vec<String>> {
+    use std::fs::File;
+    use std::io::prelude::*;
+    use std::io::BufReader;
+    use FileName::*;
+    match filename {
+        Local(path) => {
+            let file = File::open(path).ok()?;
+            BufReader::new(file)
+                .lines()
+                .into_iter()
+                .collect::<Result<Vec<String>, _>>()
+                .ok()
```

Comment:
This may not be exactly the source that was read because of races on the filesystem or similar issues. I recommend we retrieve the source from rustc directly using [`SourceMap::get_source_file`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/source_map/struct.SourceMap.html#method.get_source_file) and `SourceFile.src` instead

Reply:
Ok thanks for the remark

Reply:
Actually it's a good way of retrieving the source for the virtual files, I think